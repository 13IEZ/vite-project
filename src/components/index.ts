export * from './Buttons/PrimaryButton/PrimaryButton';
export * from './Modals/ActionsFormModal';
