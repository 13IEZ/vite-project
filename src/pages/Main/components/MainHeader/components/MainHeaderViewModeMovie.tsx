import React from 'react';
import dayjs from 'dayjs';
import { Grid, Container, IconButton, Box } from '@mui/material';
import { ColorEnum, StyledText, StyledImg } from 'style/style';
import SearchIcon from '@mui/icons-material/Search';
import { useTypedSelectorHook, useActions } from 'hooks';
import { IClickedItem } from 'store/types/movieTypes';

export const MainHeaderViewModeMovie = () => {
  const { clickedItem } = useTypedSelectorHook(state => state.movie) as IClickedItem;
  const { resetClickedMovie } = useActions();

  return (
    <header>
      <Container maxWidth='lg'>
        <Grid container justifyContent='space-between' sx={{ margin: '1.25rem 0 3.75rem' }}>
          <StyledText fontSize='125%' fontWeight={300} color={ColorEnum.PINK}>
            <strong>netflix</strong>roulette
          </StyledText>
          <IconButton size='large' color='inherit' onClick={() => resetClickedMovie()}>
            <SearchIcon fontSize='large' />
          </IconButton>
        </Grid>

        <Grid container gap={10} marginBottom='3.75rem' wrap='nowrap'>
          <Grid item xs={4}>
            <StyledImg src={clickedItem?.poster_path} alt='movie poster' />
          </Grid>

          <Grid item xs={8}>
            <Grid container>
              <StyledText fontSize='250%' fontWeight={300} color={ColorEnum.WHITE}>
                {clickedItem?.title}
              </StyledText>
              <Box
                sx={{
                  borderRadius: '50%',
                  border: '0.06rem solid white',
                  margin: 'auto',
                  padding: '0.5rem',
                }}
              >
                <StyledText fontSize='125%' fontWeight={300} color={ColorEnum.WHITE}>
                  {clickedItem?.vote_average}
                </StyledText>
              </Box>
            </Grid>

            <StyledText fontSize='87.5%' fontWeight={300} color={ColorEnum.GREY}>
              {clickedItem?.genres.join(', ')}
            </StyledText>

            <Grid container gap={5} marginTop='1.5rem'>
              <StyledText fontSize='150%' fontWeight={300} color={ColorEnum.PINK}>
                {dayjs(clickedItem?.release_date).year()}
              </StyledText>
              <StyledText fontSize='150%' fontWeight={300} color={ColorEnum.PINK}>
                {clickedItem?.runtime} min
              </StyledText>
            </Grid>

            <StyledText fontSize='125%' fontWeight={300} color={ColorEnum.GREY}>
              {clickedItem?.overview}
            </StyledText>
          </Grid>
        </Grid>
      </Container>
    </header>
  );
};
