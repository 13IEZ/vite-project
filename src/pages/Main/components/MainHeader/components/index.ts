export * from 'pages/Main/components/MainHeader/components/MainHeaderViewModeFilm';
export * from 'pages/Main/components/MainHeader/components/MainHeaderSearchbar';
