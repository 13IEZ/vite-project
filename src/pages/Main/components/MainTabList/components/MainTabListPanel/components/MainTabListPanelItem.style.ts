import styled from '@emotion/styled';
import { Grid, IconButton } from '@mui/material';
import { ColorEnum } from 'style/style';

export const StyledListPanelItem = styled(Grid)`
  transition: transform 0.5s;
  position: relative;

  &:hover {
    transform: scale(1.02);
  }
`;

export const StyledVertButton = styled(IconButton)`
  position: absolute;
  top: 0.5rem;
  right: 0.5rem;
  background-color: ${ColorEnum.DARK};
  opacity: 0.7;
`;
