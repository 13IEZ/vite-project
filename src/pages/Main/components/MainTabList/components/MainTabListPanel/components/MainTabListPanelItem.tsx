import React from 'react';
import { Grid, Menu, MenuItem } from '@mui/material';
import MoreVertIcon from '@mui/icons-material/MoreVert';
import { StyledText, ColorEnum, StyledImg } from 'style/style';
import { StyledListPanelItem, StyledVertButton } from './MainTabListPanelItem.style';
import dayjs from 'dayjs';
import { IMovie, GetMovieType } from 'store/types/movieTypes';
import { useSearchParams } from 'react-router-dom';
import { useActions } from 'hooks';

interface IMainTabListProps {
  item: IMovie;
}

const MainTabListPanelItem: React.FC<IMainTabListProps> = ({ item }) => {
  const { title, poster_path, genres, release_date, id } = item;
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const [searchParams, setSearchParams] = useSearchParams();
  const { deleteMovie, getMovie } = useActions();
  const isOpen = Boolean(anchorEl);

  const handleOpenMenu = (event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleCloseMenu = () => {
    setAnchorEl(null);
  };

  const handleOpenModal = (id: number) => {
    searchParams.set('isShowModal', id?.toString() || 'new');
    setSearchParams(searchParams);
    setAnchorEl(null);
  };

  const handleDeleteMovie = (id: number) => {
    deleteMovie(id.toString(), searchParams.toString());
    setAnchorEl(null);
  };

  return (
    <StyledListPanelItem id={'test'}>
      <Grid onClick={() => id && getMovie(id.toString(), GetMovieType.VIEW)}>
        <StyledImg src={poster_path} alt='movie poster' />
      </Grid>

      <Grid
        container
        justifyContent='space-between'
        alignItems='start'
        marginTop='1rem'
        wrap='nowrap'
      >
        <Grid item>
          <StyledText fontSize='112.5%' fontWeight={500} color={ColorEnum.GREY}>
            {title}
          </StyledText>
          <StyledText fontSize='87.5%' fontWeight={500} color={ColorEnum.DARKER_GREY}>
            {genres.join(', ')}
          </StyledText>
        </Grid>
        <Grid
          item
          padding='0.25rem'
          sx={{ border: `0.1rem solid ${ColorEnum.DARKER_GREY}`, borderRadius: '0.25rem' }}
        >
          <StyledText fontSize='87.5%' fontWeight={500} color={ColorEnum.GREY}>
            {dayjs(release_date).year()}
          </StyledText>
        </Grid>
      </Grid>
      <StyledVertButton onClick={handleOpenMenu}>
        <MoreVertIcon color='action' />
      </StyledVertButton>
      <Menu
        id='basic-menu'
        MenuListProps={{
          'aria-labelledby': 'basic-button',
        }}
        anchorEl={anchorEl}
        open={isOpen}
        onClose={handleCloseMenu}
      >
        <MenuItem onClick={() => id && handleDeleteMovie(id)}>Delete</MenuItem>
        <MenuItem onClick={() => id && handleOpenModal(id)}>Edit</MenuItem>
      </Menu>
    </StyledListPanelItem>
  );
};

export default MainTabListPanelItem;
