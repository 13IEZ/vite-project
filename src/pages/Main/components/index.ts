export * from 'pages/Main/components/MainFooter/MainFooter';
export * from 'pages/Main/components/MainHeader/MainHeader';
export * from 'pages/Main/components/MainTabList/MainTabList';
