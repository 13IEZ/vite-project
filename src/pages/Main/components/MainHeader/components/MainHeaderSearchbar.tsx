import React, { useState } from 'react';
import { Grid } from '@mui/material';
import { StyledInput } from 'style/style';
import { PrimaryButton } from 'components';
import { useActions } from 'hooks';
import { useSearchParams } from 'react-router-dom';

export const MainHeaderSearchbar = () => {
  const [value, setValue] = useState('');
  const [searchParams] = useSearchParams();
  const { getMovies } = useActions();

  const handleClickSearch = () => {
    //в апи невозможно искать по названию и сохранить фильтра по жанру
    getMovies(value ? `searchBy=title&search=${value}` : searchParams.toString());
  };

  return (
    <Grid container alignSelf='center' spacing={3} marginTop='2.5rem' paddingBottom='9.375rem'>
      <Grid item xs={9.5}>
        <StyledInput
          value={value}
          onChange={e => setValue(e.target.value)}
          label='What do you want to watch?'
          variant='outlined'
          id='test-id-input'
          fullWidth
        />
      </Grid>
      <Grid item xs={2.5}>
        <PrimaryButton
          clickHandler={handleClickSearch}
          title='search'
          type='contained'
          isFullWidth
          id='test-id-btn'
        />
      </Grid>
    </Grid>
  );
};
