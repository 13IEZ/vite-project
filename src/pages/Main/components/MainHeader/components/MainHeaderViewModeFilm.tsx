import React, { Dispatch, SetStateAction } from 'react';
import dayjs from 'dayjs';
import { Grid, Container, IconButton, Box } from '@mui/material';
import { IIMainTabListPanelItem } from 'pages/Main/components/MainTabList/components/MainTabListPanel/components/MainTabListPanelItem';
import { ColorEnum, StyledText, StyledImg } from 'style/style';
import SearchIcon from '@mui/icons-material/Search';

interface IMainHeaderViewModeFilm {
  clickedItemToView: null | IIMainTabListPanelItem;
  setClickedItemToView: Dispatch<SetStateAction<IIMainTabListPanelItem | null>>;
}

export const MainHeaderViewModeFilm: React.FC<IMainHeaderViewModeFilm> = ({
  clickedItemToView,
  setClickedItemToView,
}) => {
  return (
    <header>
      <Container maxWidth='lg'>
        <Grid container justifyContent='space-between' sx={{ margin: '1.25rem 0 3.75rem' }}>
          <StyledText fontSize='125%' fontWeight={300} color={ColorEnum.PINK}>
            <strong>netflix</strong>roulette
          </StyledText>
          <IconButton size='large' color='inherit' onClick={() => setClickedItemToView(null)}>
            <SearchIcon fontSize='large' />
          </IconButton>
        </Grid>

        <Grid container gap={10} marginBottom='3.75rem' wrap='nowrap'>
          <Grid item xs={4}>
            <StyledImg src={clickedItemToView?.image} alt='movie poster' />
          </Grid>

          <Grid item xs={8}>
            <Grid container>
              <StyledText fontSize='250%' fontWeight={300} color={ColorEnum.WHITE}>
                {clickedItemToView?.title}
              </StyledText>
              <Box
                sx={{
                  borderRadius: '50%',
                  border: '0.06rem solid white',
                  margin: 'auto',
                  padding: '0.5rem',
                }}
              >
                <StyledText fontSize='125%' fontWeight={300} color={ColorEnum.WHITE}>
                  {clickedItemToView?.rating}
                </StyledText>
              </Box>
            </Grid>

            <StyledText fontSize='87.5%' fontWeight={300} color={ColorEnum.GREY}>
              {clickedItemToView?.genre}
            </StyledText>

            <Grid container gap={5} marginTop='1.5rem'>
              <StyledText fontSize='150%' fontWeight={300} color={ColorEnum.PINK}>
                {dayjs(clickedItemToView?.year).year()}
              </StyledText>
              <StyledText fontSize='150%' fontWeight={300} color={ColorEnum.PINK}>
                {clickedItemToView?.runtime} min
              </StyledText>
            </Grid>

            <StyledText fontSize='125%' fontWeight={300} color={ColorEnum.GREY}>
              {clickedItemToView?.overview}
            </StyledText>
          </Grid>
        </Grid>
      </Container>
    </header>
  );
};
