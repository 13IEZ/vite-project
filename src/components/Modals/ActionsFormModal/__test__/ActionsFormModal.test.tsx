import React from 'react';
import { expect, it, describe } from 'vitest';
import store from 'store/configureStore';
import { ActionsFormModal } from 'components';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

describe('ActionsFormModal', () => {
  it('should render modal', () => {
    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <ActionsFormModal isOpen={true} />
        </MemoryRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
