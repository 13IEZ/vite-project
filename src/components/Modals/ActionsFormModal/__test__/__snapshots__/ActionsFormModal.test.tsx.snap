// Vitest Snapshot v1

exports[`ActionsFormModal > should render modal 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiModal-root css-79ws1d-MuiModal-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="css-j9aoto MuiBox-root css-0"
        tabindex="-1"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-1svu6jh-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 css-10n7unq css-vn7k65-MuiTypography-root"
        >
          ADD MOVIE
        </p>
        <form>
          <div
            class="MuiGrid-root MuiGrid-container css-16x50u-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-446am0 css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="tittle"
                  id="tittle-label"
                >
                  TITLE
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="tittle"
                    name="title"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        TITLE
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-446am0 css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="poster_path"
                  id="poster_path-label"
                >
                  MOVIE URL
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="poster_path"
                    name="poster_path"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        MOVIE URL
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-446am0 css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for=":r2:"
                  id=":r2:-label"
                >
                  GENRES
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby=":r2:-label :r2:"
                    class="MuiSelect-select MuiSelect-outlined MuiSelect-multiple MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r2:"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    name="genres"
                    tabindex="-1"
                    value="Action,Test"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        GENRES
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-446am0 css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r3:"
                    name="release_date"
                    type="date"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-446am0 css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for=":r4:"
                  id=":r4:-label"
                >
                  RATING
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r4:"
                    maxlength="2"
                    name="vote_average"
                    type="number"
                    value="0"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        RATING
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-446am0 css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for=":r5:"
                  id=":r5:-label"
                >
                  RUNTIME
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r5:"
                    name="runtime"
                    type="number"
                    value="0"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        RUNTIME
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-446am0 css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for=":r6:"
                  id=":r6:-label"
                >
                  OVERVIEW
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <textarea
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r6:"
                    name="overview"
                    style="height: 0px; overflow: hidden;"
                  />
                  <textarea
                    aria-hidden="true"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                    readonly=""
                    style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                    tabindex="-1"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        OVERVIEW
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container css-1jc6cju-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth css-z5heq2 css-xsqe0f-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Reset
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-cv1qiv css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Save
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
